### ISSUE TITLE
Confusing docs...

### ISSUE THREAD

**Issue by j** [Automattic/mongoose#1538]
It's silly for me to write an issue on this, so this will most likely get closed pretty quickly, but I have a question regarding:

http://mongoosejs.com/docs/api.html#document_Document-toObject

The line states:
`Converts this document into a plain javascript object, ready for storage in MongoDB.`

but later in the options docs it says:
`transform a transform function to apply to the resulting document before returning`

So it gives me the vibe that before inserting the object into the database, it calls this method.  But the doc examples make it appear to just transform an object for representation only... Does mongoose internally use this for persistance of the object also?  I haven't had time to dig around into this.

Basically, I want to know if I can use a global schema transformer for the REST output and to not manipulate the mongo document in any way.  Is this where I do it?

Or should I have to pass the transformer into the toObject method where I need it.
